Add bootloader emulator builds into CI

They are not being used very often, so it can happen that we break them without noticing - and then they do not work when we need them.

Is the bootloader emulator supposed to work for model T? I'm getting
> secret.c:25: undefined reference to `SECRET_AREA'

https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/jobs/4993757912

--- a/ci/build.yml
+++ b/ci/build.yml
@@ -214,10 +214,12 @@ core unix regular asan build:
   variables:
     ADDRESS_SANITIZER: "1"
   script:
+    - $NIX_SHELL --run "poetry run make -C core build_bootloader_emu"
     - $NIX_SHELL --run "poetry run make -C core build_unix"
   artifacts:
     name: "$CI_JOB_NAME-$CI_COMMIT_SHORT_SHA"
     paths:
+      - core/build/bootloader_emu/bootloader.elf
       - core/build/unix  # most of it needed by test_rust
     expire_in: 1 week
 
@@ -229,10 +231,12 @@ core unix frozen regular build:
   <<: *gitlab_caching
   needs: []
   script:
+    - $NIX_SHELL --run "poetry run make -C core build_bootloader_emu"
     - $NIX_SHELL --run "poetry run make -C core build_unix_frozen"
   artifacts:
     name: "$CI_JOB_NAME-$CI_COMMIT_SHORT_SHA"
     paths:
+      - core/build/bootloader_emu/bootloader.elf
       - core/build/unix/trezor-emu-core
     expire_in: 1 week
 
@@ -301,7 +305,8 @@ core unix frozen R debug build:
     PYOPT: "0"
     TREZOR_MODEL: "R"
   script:
-    - nix-shell --run "poetry run make -C core build_unix_frozen"
+    - $NIX_SHELL --run "poetry run make -C core build_bootloader_emu"
+    - $NIX_SHELL --run "poetry run make -C core build_unix_frozen"
   artifacts:
     name: "$CI_JOB_NAME-$CI_COMMIT_SHORT_SHA"
     untracked: true
@@ -316,7 +321,7 @@ core unix frozen R debug build arm:
     PYOPT: "0"
     TREZOR_MODEL: "R"
   script:
-    - nix-shell --run "poetry run make -C core build_unix_frozen"
+    - $NIX_SHELL --run "poetry run make -C core build_unix_frozen"
     - mv core/build/unix/trezor-emu-core core/build/unix/trezor-emu-core-arm
   artifacts:
     name: "$CI_JOB_NAME-$CI_COMMIT_SHORT_SHA"

--- a/core/SConscript.bootloader_emu
+++ b/core/SConscript.bootloader_emu
@@ -128,8 +128,12 @@ SOURCE_TREZORHAL = [
     'embed/trezorhal/unix/rng.c',
     'embed/trezorhal/unix/usb.c',
     'embed/trezorhal/unix/random_delays.c',
-    'embed/trezorhal/unix/secret.c',
 ]
+
+if TREZOR_MODEL in ('R', ):
+    SOURCE_TREZORHAL += [
+        'embed/trezorhal/unix/secret.c',
+    ]
 
 SOURCE_UNIX = [
     'embed/unix/profile.c',

--- a/docs/ci/jobs.md
+++ b/docs/ci/jobs.md
@@ -83,60 +83,60 @@ present which get interpreted.
 
 ### [core unix regular asan build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L208)
 
-### [core unix frozen regular build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L227)
+### [core unix frozen regular build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L229)
 Build of Core into UNIX emulator. Something you can run on your laptop.
 Frozen version. That means you do not need any other files to run it,
 it is just a single binary file that you can execute directly.
 
-### [core unix frozen btconly debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L244)
+### [core unix frozen btconly debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L248)
 Build of Core into UNIX emulator. Something you can run on your laptop.
 Frozen version. That means you do not need any other files to run it,
 it is just a single binary file that you can execute directly.
 See [Emulator](../core/emulator/index.md) for more info.
 Debug mode enabled, Bitcoin-only version.
 
-### [core unix frozen btconly debug asan build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L260)
+### [core unix frozen btconly debug asan build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L264)
 
-### [core unix frozen debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L283)
+### [core unix frozen debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L287)
 Build of Core into UNIX emulator. Something you can run on your laptop.
 Frozen version. That means you do not need any other files to run it,
 it is just a single binary file that you can execute directly.
 **Are you looking for a Trezor T emulator? This is most likely it.**
 
-### [core unix frozen R debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L296)
+### [core unix frozen R debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L300)
 
-### [core unix frozen R debug build arm](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L310)
+### [core unix frozen R debug build arm](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L315)
 
-### [core unix frozen debug asan build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L328)
+### [core unix frozen debug asan build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L333)
 
-### [core unix frozen debug build arm](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L344)
+### [core unix frozen debug build arm](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L349)
 
-### [core macos frozen regular build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L366)
+### [core macos frozen regular build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L371)
 
-### [crypto build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L391)
+### [crypto build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L396)
 Build of our cryptographic library, which is then incorporated into the other builds.
 
-### [legacy fw regular build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L421)
+### [legacy fw regular build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L426)
 
-### [legacy fw regular debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L437)
+### [legacy fw regular debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L442)
 
-### [legacy fw btconly build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L454)
+### [legacy fw btconly build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L459)
 
-### [legacy fw btconly debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L473)
+### [legacy fw btconly debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L478)
 
-### [legacy emu regular debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L494)
+### [legacy emu regular debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L499)
 Regular version (not only Bitcoin) of above.
 **Are you looking for a Trezor One emulator? This is most likely it.**
 
-### [legacy emu regular debug asan build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L509)
+### [legacy emu regular debug asan build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L514)
 
-### [legacy emu regular debug build arm](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L527)
+### [legacy emu regular debug build arm](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L532)
 
-### [legacy emu btconly debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L553)
+### [legacy emu btconly debug build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L558)
 Build of Legacy into UNIX emulator. Use keyboard arrows to emulate button presses.
 Bitcoin-only version.
 
-### [legacy emu btconly debug asan build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L570)
+### [legacy emu btconly debug asan build](https://github.com/trezor/trezor-firmware/blob/master/ci/build.yml#L575)
 
 ---
 ## TEST stage - [test.yml](https://github.com/trezor/trezor-firmware/blob/master/ci/test.yml)

